src/test/java/uk/gov/justice/hmpps/prison/api/resource/impl/OffenderResourceImplIntTestGetOffenderBeliefHistory.kt

```diff
@@ -0,0 +1,81 @@
+package uk.gov.justice.hmpps.prison.api.resource.impl
+
+import org.junit.jupiter.api.Test
+import org.mockito.ArgumentMatchers.anyString
+import org.mockito.kotlin.anyOrNull
+import org.mockito.kotlin.whenever
+import org.springframework.boot.test.mock.mockito.MockBean
+import org.springframework.http.HttpMethod.GET
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderBelief
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderBooking
+import uk.gov.justice.hmpps.prison.repository.jpa.model.ProfileCode
+import uk.gov.justice.hmpps.prison.repository.jpa.model.ProfileType
+import uk.gov.justice.hmpps.prison.repository.jpa.model.Staff
+import uk.gov.justice.hmpps.prison.repository.jpa.model.StaffUserAccount
+import uk.gov.justice.hmpps.prison.repository.jpa.repository.OffenderBeliefRepository
+import java.time.LocalDateTime
+
+class OffenderResourceImplIntTestGetOffenderBeliefHistory : ResourceTest() {
+
+  private val beliefCode = ProfileCode.builder().id(ProfileCode.PK(ProfileType.builder().type("RELF").build(), "SCIE")).description("Scientologist").build()
+  private val staffUserAccount = StaffUserAccount.builder()
+    .username("johnsmith")
+    .staff(
+      Staff.builder()
+        .firstName("John")
+        .lastName("Smith")
+        .build(),
+    )
+    .build()
+  private val offenderBelief = OffenderBelief(
+    98765L, OffenderBooking.builder().bookingId(123456L).build(),
+    beliefCode, LocalDateTime.parse("2024-01-01T00:00:00"), null, true, "Comments",
+    false, LocalDateTime.parse("2024-01-01T00:00:00"), staffUserAccount, null, null,
+  )
+  private fun stubRepositoryCall() {
+    whenever(offenderBeliefRepository.getOffenderBeliefHistory(anyString(), anyOrNull())).thenReturn(
+      listOf(
+        offenderBelief,
+      ),
+    )
+  }
+
+  @MockBean
+  private lateinit var offenderBeliefRepository: OffenderBeliefRepository
+
+  @Test
+  fun `should return 401 when user does not even have token`() {
+    webTestClient.get().uri("/api/offenders/B1101BB/belief-history")
+      .exchange()
+      .expectStatus().isUnauthorized
+  }
+
+  @Test
+  fun `should return 403 if no override role`() {
+    webTestClient.get().uri("/api/offenders/B1101BB/belief-history")
+      .headers(setClientAuthorisation(listOf()))
+      .exchange()
+      .expectStatus().isForbidden
+  }
+
+  @Test
+  fun `should return success when has ROLE_VIEW_PRISONER_DATA override role`() {
+    stubRepositoryCall()
+    webTestClient.get().uri("/api/offenders/B1101BB/belief-history")
+      .headers(setClientAuthorisation(listOf("ROLE_VIEW_PRISONER_DATA")))
+      .exchange()
+      .expectStatus().isOk
+  }
+
+  @Test
+  fun shouldReturnOffenderBeliefHistory() {
+    stubRepositoryCall()
+    val responseEntity = testRestTemplate.exchange(
```

Comment:
better to stick with using webTestClient (we are moving away from testRestClient)